dhall-json/tasty/Main.hs

```diff
@@ -142,3 +147,38 @@ testJSONToDhall =
 inferJSONToDhall :: String -> TestTree
 inferJSONToDhall =
     testCustomConversionJSONToDhall True JSONToDhall.defaultConversion
+
+loadSchemaFromFile :: FilePath -> IO (Core.Expr Dhall.Parser.Src Void)
+loadSchemaFromFile schemaFile = do
+    schemaText <- Data.Text.IO.readFile schemaFile
+
+    parsedSchema <- Core.throws (Dhall.Parser.exprFromText schemaFile schemaText)
+
+    Dhall.Import.load parsedSchema
+
+testJSONToDhallErrorMessage :: String -> JSONToDhall.Conversion -> TestTree
+testJSONToDhallErrorMessage prefix conv =
+    Test.Tasty.Silver.goldenVsAction prefix goldenFile action converter
```

Comment:
Thanks for giving my suggestion for `tasty-silver` a try! :)

I had never used it before, so I just checked whether the automatic updating of golden files with the `--accept` option works:

```
$ cabal run dhall-json:tasty -w ghc-8.6.5 -- --accept
```

Note that you have to be in the `dhall-json` directory for this, otherwise you get errors like

```
Exception: ./tasty/data/mismatchMessage0.json: openBinaryFile: does not exist (No such file or directory)
```

Not sure whether this is something that could be fixed in `tasty-silver`.

Anyway, it's really cool, and maybe we can adopt this for other tests like those for `dhall format` and `dhall lint`.

Reply:
> Not sure whether this is something that could be fixed in `tasty-silver`.

Nope, I just need to use `cabal test` with `--test-option` instead of `cabal run`! :)

This works in the project root:

```
$ cabal test dhall-json:tasty -w ghc-8.6.5 --test-option --accept
```

Reply:
Oh I didn't knew about those automatic updates of golden files. I'll give it a try.

Although that option comes handy, I think that we should try to avoid updating those golden files unless necessasry.

Reply:
Yeah. This would be really useful to have for the acceptance test suite or e.g. the formatting tests. It's really handy to be able to just use `--accept` and then look at the differences with `git diff`.

Reply:
Yeah, this would be very useful for other tests, since we have a large number of golden tests across the project